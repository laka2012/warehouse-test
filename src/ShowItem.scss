.showitem{
    display: inline-block;
    padding: 15px;
    margin: 5px;
    border: 1px solid #b4bac4;
    .name {
        border-bottom: 1px solid #b4bac4;
        padding-bottom: 5px;
        text-align: center;
        width: 300px;
    }
    .image {
        margin: 10px 0px;
        img {
            width: 300px;
        }
    }
    .bottom {
        border-top: 1px solid #b4bac4;
        padding: 5px;
        .price {
            float: left;
            margin-left: 20px;
        }
        .add-cart {
            float: right;
            margin-right: 20px;
        }
    }
    &:hover {
        box-shadow: 0 0 20px #a4c2f2; 
    }
}