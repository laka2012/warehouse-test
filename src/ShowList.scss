.showlist {
    margin-left: 220px;
    padding: 0px 20px;
    .pagingbar {
        text-align: center;
    }
    .search-desc {
        margin: 10px 5px;
        font-size: large;
        font-weight: bold;
    }
}

#react-paginate ul {
    display: inline-block;
    padding-left: 15px;
    padding-right: 15px;
}

#react-paginate li {
    display: inline-block;
}

@media screen and (max-width: 800px) {
    .showlist {
        margin-left: 0px;
    }
}