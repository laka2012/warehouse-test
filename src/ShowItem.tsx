import React, { Component } from 'react';
import './ShowItem.scss';
import Item from './IItem';

export interface Props {
	item: Item;
	addToCart: (item: Item) => void; 
}

class ShowItem extends Component<Props> {
	constructor(props: Props) {
		super(props);
		this.state = {isToggleOn: true};
		this.handleAddCartClick = this.handleAddCartClick.bind(this);
	}

	handleAddCartClick() {
		this.props.addToCart(this.props.item);
	}
	
	render() {
		return (
			<div className='showitem'>
				<div className='name'>{this.props.item.name}</div>
				<div className='image'>
					<img src={this.props.item.image_url} ></img>
				</div>
				<div className='bottom'>
					<div className='price'>${this.props.item.price}</div>
					<button className='add-cart' onClick={this.handleAddCartClick}>Add To Cart</button>
				</div>
			</div>
		);
	}
}

export default ShowItem;
