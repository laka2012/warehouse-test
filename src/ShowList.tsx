import React, { Component } from 'react';
import './ShowList.scss';
import ShowItem from './ShowItem';
import ReactPaginate from 'react-paginate';
import Item from './IItem';

export interface Props {
    items: Item[];
    pageCount: number;
    handlePageClick: (data:any) => void;
    addToCart: (item: Item) => void;
    searchText: string;
}

class ShowList extends Component<Props> {
    constructor(props: Props) {
        super(props);
        this.onPageChange = this.onPageChange.bind(this);
    }

    onPageChange(data: any) {
        this.props.handlePageClick(data);
    }

    render() {
        return (
            <div className='showlist'>
                { this.props.searchText != '' &&
                    <div className='search-desc'>You are searching "{this.props.searchText}" in name or brand</div>
                }
                <div className='items'>
                    {this.props.items.map((item: Item) =>
                        <ShowItem key={item.sku} item={item} addToCart={this.props.addToCart} />
                    )}
                </div>
                <div className='pagingbar'>
                    <ReactPaginate
                        previousLabel={'previous'}
                        nextLabel={'next'}
                        breakLabel={'...'}
                        breakClassName={'break-me'}
                        pageCount={this.props.pageCount}
                        marginPagesDisplayed={2}
                        pageRangeDisplayed={5}
                        onPageChange={this.onPageChange}
                        containerClassName={'pagination'}
                        activeClassName={'active'}
                    />
                </div>
            </div>
        );
    }
}

export default ShowList;
