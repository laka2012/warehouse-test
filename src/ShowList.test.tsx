import React from 'react';
import { shallow } from 'enzyme';
import ShowList, {Props} from './ShowList';
import Item from './IItem';
import ShowItem from './ShowItem';

function showListTest(props: Partial<Props> = {}) {
    return (
        <ShowList {...{
            items: [],
            pageCount: 10,
            handlePageClick: (data: any) => {},
            addToCart: (item: Item) => {},
            searchText: '',
            ...props}}>
        </ShowList>
    );
}

describe('ShowList', () => {
	it('smoke test', () => {
		const wrapper = shallow(
			showListTest()
		);
		expect(wrapper).toMatchSnapshot();
    });

    it('search description test', () => {
		const wrapper = shallow(
			showListTest({searchText: 'test'})
		);
		expect(wrapper.find('.search-desc')).toHaveLength(1);
    });
    
    it('show ShowItem test', () => {
		const wrapper = shallow(
			showListTest({items: [{
                date_added: 'test',
                type: 'test',
                sku: 'test',
                group_id: 'test',
                name: 'test',
                model: 'test',
                brand: 'test',
                url: 'test',
                image_url: 'test',
                on_special: true,
                price: 100,
                flybuys_points: 1005,
                shipping_class: 'test',
                add_to_cart: true,
                categories: 'test',
                in_stock: true,
                sold_out: false}]})
		);
		expect(wrapper.find(ShowItem)).toHaveLength(1);
	});
})