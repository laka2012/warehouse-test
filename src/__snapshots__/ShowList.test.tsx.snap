// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowList smoke test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowList
    addToCart={[Function]}
    handlePageClick={[Function]}
    items={Array []}
    pageCount={10}
    searchText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="items"
        />,
        <div
          className="pagingbar"
        >
          <PaginationBoxView
            activeClassName="active"
            breakClassName="break-me"
            breakLabel="..."
            containerClassName="pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={2}
            nextClassName="next"
            nextLabel="next"
            onPageChange={[Function]}
            pageCount={10}
            pageRangeDisplayed={5}
            previousClassName="previous"
            previousLabel="previous"
          />
        </div>,
      ],
      "className": "showlist",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "items",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PaginationBoxView
            activeClassName="active"
            breakClassName="break-me"
            breakLabel="..."
            containerClassName="pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={2}
            nextClassName="next"
            nextLabel="next"
            onPageChange={[Function]}
            pageCount={10}
            pageRangeDisplayed={5}
            previousClassName="previous"
            previousLabel="previous"
          />,
          "className": "pagingbar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeClassName": "active",
            "breakClassName": "break-me",
            "breakLabel": "...",
            "containerClassName": "pagination",
            "disableInitialCallback": false,
            "disabledClassName": "disabled",
            "marginPagesDisplayed": 2,
            "nextClassName": "next",
            "nextLabel": "next",
            "onPageChange": [Function],
            "pageCount": 10,
            "pageRangeDisplayed": 5,
            "previousClassName": "previous",
            "previousLabel": "previous",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="items"
          />,
          <div
            className="pagingbar"
          >
            <PaginationBoxView
              activeClassName="active"
              breakClassName="break-me"
              breakLabel="..."
              containerClassName="pagination"
              disableInitialCallback={false}
              disabledClassName="disabled"
              marginPagesDisplayed={2}
              nextClassName="next"
              nextLabel="next"
              onPageChange={[Function]}
              pageCount={10}
              pageRangeDisplayed={5}
              previousClassName="previous"
              previousLabel="previous"
            />
          </div>,
        ],
        "className": "showlist",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "items",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PaginationBoxView
              activeClassName="active"
              breakClassName="break-me"
              breakLabel="..."
              containerClassName="pagination"
              disableInitialCallback={false}
              disabledClassName="disabled"
              marginPagesDisplayed={2}
              nextClassName="next"
              nextLabel="next"
              onPageChange={[Function]}
              pageCount={10}
              pageRangeDisplayed={5}
              previousClassName="previous"
              previousLabel="previous"
            />,
            "className": "pagingbar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "breakClassName": "break-me",
              "breakLabel": "...",
              "containerClassName": "pagination",
              "disableInitialCallback": false,
              "disabledClassName": "disabled",
              "marginPagesDisplayed": 2,
              "nextClassName": "next",
              "nextLabel": "next",
              "onPageChange": [Function],
              "pageCount": 10,
              "pageRangeDisplayed": 5,
              "previousClassName": "previous",
              "previousLabel": "previous",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
